components/ws-manager-bridge/src/bridge.ts

```diff
@@ -166,13 +167,52 @@ export class WorkspaceManagerBridge implements Disposable {
     }
 
     protected async handleStatusUpdate(ctx: TraceContext, rawStatus: WorkspaceStatus, writeToDB: boolean) {
+        const start = performance.now();
         const status = rawStatus.toObject();
         log.info("Handling WorkspaceStatus update", status);
+
         if (!status.spec || !status.metadata || !status.conditions) {
             log.warn("Received invalid status update", status);
             return;
         }
 
+        try {
+            this.prometheusExporter.reportWorkspaceInstanceUpdateStarted(
+                writeToDB,
+                this.cluster.name,
+                status.spec.type,
+            );
+            await this.statusUpdate(ctx, rawStatus, writeToDB);
+        } catch (e) {
+            const durationMs = performance.now() - start;
+            this.prometheusExporter.reportWorkspaceInstanceUpdateCompleted(
+                durationMs / 1000,
+                writeToDB,
+                this.cluster.name,
+                status.spec.type,
+                e,
+            );
+            log.error("Failed to complete WorkspaceInstance status update", e);
+            throw e;
+        } finally {
+            const durationMs = performance.now() - start;
+            this.prometheusExporter.reportWorkspaceInstanceUpdateCompleted(
+                durationMs / 1000,
+                writeToDB,
+                this.cluster.name,
+                status.spec.type,
+            );
+            log.info("Successfully completed WorkspaceInstance status update");
```

Comment:
What about adding owi/logcontext to these log lines (like [here](https://github.com/gitpod-io/gitpod/pull/9422/files#diff-5c377b2cad37712ee6f512a499f1069f7641af5aab7dceaa7e8d033fc23d4780L197))?

Reply:
Added.